### ISSUE TITLE
feature request: conditional property management

### ISSUE THREAD

**Issue by weeyizhi** [mdelobelle/obsidian_supercharged_links#8]
I think it would be really cool if we could set up conditional property management.
For example, the `status` property of a file with `type: task`  might be able to contain `new`, `todo`, `done`, etc. But for a note, it might contain `new`, `in-progress`, `reviewed`, etc.

Ideally (for me), the default behavior should be just one 'global' rule, but user can add more specific conditions with their own rules as needed for said property.

**Comment by mdelobelle** [comment]
hello
this now possible with 0.2.5

**Comment by mdelobelle** [comment]
You can find some detail in readMe at section "define a class for a file and authorized fields for this class"

**Comment by weeyizhi** [comment]
Hi! Thanks for the amazing work. I think there is a slight misunderstanding here, but we're almost there. The new feature does bring us one step closer.

Let's say I have a file of `fileClass: task`
```md
task.md
=========
status
```
And another file of `fileClass: note`
```md
note.md
=========
status
```
I would like files of different classes to be able to contain different values inside their `status` field. My suggestion would be to specify possible values for classes as well. If such specification is done, override the global property manager settings.

Eg.
```md
task.md
=========
status: [new, to-do, in-progress, cancelled, done, on-hold]
```

```md
note.md
=========
status: [new, in-progress, reviewed]
```
Let me know if you think this would be a useful feature.

**Comment by mdelobelle** [comment]
@weeyizhi I get your point.
Now that we can manage which field for which category of note, it shouldn't be too hard to define presets values for them. Just like in the settings where you can manage them globally, you could have those preset values in the "class file" for a category.
I'll have a look at that in the coming days

**Comment by mdelobelle** [comment]
Hi this is now possible with version 0.2.9. Does it correspond with your request?

**Comment by weeyizhi** [comment]
Hi. I tried for quite a while and could not get it to work. Then I noticed that somehow I'm unable to update to 0.2.9 via community plugins.

Clicking 'update' shows this (but the installed version only went from 0.2.7 to 0.2.8). Doing it again shows the same thing (installed version stays at 0.2.8). Reinstall also gives me the same result.
![image](https://user-images.githubusercontent.com/7296340/120571913-ac77f280-c44d-11eb-9135-20602db8bd4f.png)

**Comment by mdelobelle** [comment]
What is your obsidian version ? The minimal obsidian version for 0.2.9 is 0.12.4

**Comment by weeyizhi** [comment]
Ah I see. I'm still at 0.12.3 because I'm not an insider! Will give this a go again when I get the update!

**Comment by mdelobelle** [comment]
you should be able to get 0.2.10 that is available with 0.12.3 now

**Comment by weeyizhi** [comment]
It works amazingly! The only comment I have is when we click on `Manage <file class name> fields`, instead of saying hello, it might be good to mention the fileClass name instead:
![image](https://user-images.githubusercontent.com/7296340/120594884-24f1aa00-c474-11eb-83ac-618c6d5d514d.png)

**Comment by mdelobelle** [comment]
good catch! 😉
fixed in 0.2.11
I'm glad that you enjoy the feature, it's very helpful for my workflow too, thanks for the idea
